xbmc/addons/BinaryAddonCache.h

```diff
@@ -31,8 +31,9 @@ namespace ADDON {
 class CBinaryAddonCache : public Observer
 {
 public:
-  virtual ~CBinaryAddonCache();
+  virtual ~CBinaryAddonCache() { }
```

Comment:
> Init() call should have a corresponding Deinit()

You put something in an Init method because it can fail and those type of things do not fit into a constructor. THe destructor is there for doing cleanup, no matter if something fails.